index.js

```diff
@@ -60,9 +60,11 @@ function transform(rule) {
     return
   }
 
+  // In PostCSS version 8.4.6, this will contain comments that need to be removed manually
+  rule.params = rule.params.replace(/\/\*.*\*\//gs, '')
```

Comment:
We found a few rare cases when comment removing went to meaning changes. We do not remove this comment (we didn’t remove these comments from `Declaration#value` too, so it is part of API, we just add more use cases to it)

Reply:
Sorry, I didn't understand.

In the Change Log for 8.4.6 I see: 

> Prevented comment removing when it change meaning of CSS.

Is this change the cause?

If the user is not using npm lock, does it cause an error with this plugin? How do I avoid incompatibility issues with PostCSS MINOR version upgrades?

Reply:
> Is this change the cause?

Yes.

Here is an example:

```css
@some a/**/b {}
```

It has 2 tokens `a` and `b` (like `@some a b {}`). If we remove comment, it will become `@some ab {}`, which is completly another param.

So in 8.4.6 we fixed the bug.

Reply:
So it looks like we need a new `postcss-at-rule-parser` parsing plugin, right?

As the CSS at rule syntax gets more complex, the current regular pattern is too painful to maintain.

Reply:
Yes, it is a good idea 😄

Reply:
> So it looks like we need a new postcss-at-rule-parser parsing plugin, right?

@yisibl Is anyone working on this? We have a need for this too but only `@media` would be sufficient for us.

Unless there is a package that is nearing completion we will create something that can parse [`MediaQueryList`](https://drafts.csswg.org/mediaqueries-5/#mq-syntax) with PostCSS as inspiration for the API surface.

I currently have time to spend on this but obviously prefer to use an existing package if one exists :)

Reply:
WIP: https://github.com/postcss/postcss-at-rule-parser

Reply:
:/ I won't be able to help much with that.
I don't want to context switch to Rust and WASM at the moment.

Nice to see the progress though!